CHANGELOG.md

```diff
@@ -0,0 +1,54 @@
+- For test purposes only, it is now possible to use OTA to downgrade the version number. That capability is disabled by default.
+- Add doxygen comments to the OTA source files.
+
+### Updates
+
+#### Rename FreeRTOS-IoT-Libraries Directory
+
+- The FreeRTOS-IoT-Libraries directories in the previous snapshot version (V2020218_LTS_development_snapshot) under FreeRTOS-Plus/Source and FreeRTOS-Plus/Demo has been renamed to *FreeRTOS-IoT-Libraries-Beta1* to make way for a parallel directory, *FreeRTOS-IoT-Libraries-Beta2*, that holds libraries for which the refactoring work is near completion (currently the MQTT library).
+
+#### New MQTT Client Library (Beta 2)
+
+- A new MQTT library source code has been added under the FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-Beta2 directory.
+- Demos for the new MQTT library have been added to the FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-Beta2 directory.
+
+#### Bugfixes in MQTT Client Library (Beta 1)
+
+* Improved the Keep-Alive mechanism: The MQTT library will not send PING requests when connection is not idle, which fixes a disconnect issue during OTA. In the prior version, MQTT would sometimes disconnect during OTA due to timeouts for server PING response.
+* Bug fix for Keep-Alive interval: The MQTT library was incorrectly sending PING requests at intervals greater than the keep alive period sent in the CONNECT request. This change fixes the problem.
+* Bugfix for synchronous Publish at QoS 0: The earlier implementation of IotMqtt_PublishSync  API always returned success even when network transmission of the PUBLISH packet failed. This bug has now been fixed such that the API function will return network error when network transmission fails.
+
+#### PKCS#11
+
+- Added doxygen to various PKCS #11 files.
+- Added improved logging for mbed TLS  return codes in iot_pkcs11_mbedtls.c. 
+- Added PKCS11_PAL_Initialize for initializing the PKCS #11 PAL layer.
+- Refactored various MISRA violations.
+- Refactored Sessions to use static memory instead of Dynamic memory.
+
+#### FreeRTOS+TCP 
+
+- Added ability to cache multiple IP addresses per DNS entry.
+- Defensive security improvements: 
+    - In compliance with the UDP protocol specification, prior versions of FreeRTOS+TCP accepted UDP packets that had their checksum set to 0. FreeRTOS+TCP V2.3.0 adds a new configuration parameter, ipconfigUDP_PASS_ZERO_CHECKSUM_PACKETS, that enables users to opt to drop packets UDP packets that have their checksum set to 0. Note this new setting defaults to 0, so it defaults to dropping UDP packets that have their checksum set to 0.
```

Comment:
to drop packets UDP packets --> to drop UDP packets

Reply:
Fixed